client/xmlrpc_client_test.go

```diff
@@ -66,3 +87,62 @@ func TestExecuteCall(t *testing.T) {
 		})
 	}
 }
+
+func TestExecuteCallWithTimeout(t *testing.T) {
+	tt := []struct {
+		name             string
+		connectTimeout   int
+		requestTimeout   int
+		sleepTime        time.Duration
+		expectedResponse interface{}
+		expectedError    string
+	}{
+		{name: "ExecuteCall Pass",
+			connectTimeout:   1,
+			requestTimeout:   1,
+			sleepTime:        0,
+			expectedResponse: "hello world",
+		},
+		{name: "ExecuteCall Fail",
+			connectTimeout: 1,
+			requestTimeout: 1,
+			sleepTime:      2,
+			expectedError:  "i/o timeout",
+		},
+	}
+
+	for _, tc := range tt {
+		t.Run(tc.name, func(t *testing.T) {
+			//init server
+			ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				log.Println("Sleeping...")
+				time.Sleep(tc.sleepTime * time.Second)
+				io.WriteString(w, sampleResponse)
+			}))
+			defer ts.Close()
+
+			//init client
+			client := NewClient(tc.connectTimeout, tc.requestTimeout)
+			credentials := []interface{}{"tc.username, tc.password"}
```

Comment:
why do we need the credentials?

Reply:
We don't... I was testing some different stuff and it sneaked in. I have removed it. Thank you